src/workflow/test/Util_GTest.cpp

```diff
@@ -68,3 +78,120 @@ TEST_F(WorkflowFixture, Util_sanitizeKey) {
+    newObject.setString(0, "SimpleAndTabular");
+    newObject.setString(1, "JtoKWH");
+
+    EXPECT_TRUE(openstudio::workflow::util::addEnergyPlusOutputRequest(w, newObject));
+    {
+      auto wos = w.getObjectsByType(IddObjectType::Output_SQLite);
+      ASSERT_EQ(1, wos.size());
+      const auto& new_wo = wos.front();
+      EXPECT_EQ("SimpleAndTabular", new_wo.getString(0).get());
+      EXPECT_EQ("JtoKWH", new_wo.getString(1).get());
+    }
+  }
+  {
+    // Unique and not present already
+    auto& newObject = energyPlusOutputRequests.emplace_back(IddObjectType::Output_Diagnostics);
+    newObject.setString(0, "DisplayExtraWarnings");
+    newObject.setString(1, "DisplayUnusedSchedules");
+
+    EXPECT_EQ(0, w.getObjectsByType(IddObjectType::Output_Diagnostics).size());
+    EXPECT_TRUE(openstudio::workflow::util::addEnergyPlusOutputRequest(w, newObject));
+    {
+      auto wos = w.getObjectsByType(IddObjectType::Output_Diagnostics);
+      ASSERT_EQ(1, wos.size());
+      const auto& new_wo = wos.front();
+      EXPECT_EQ("DisplayExtraWarnings", new_wo.getString(0).get());
+      EXPECT_EQ("DisplayUnusedSchedules", new_wo.getString(1).get());
+    }
+  }
+  {
+    // Non unique, and present already
+    WorkspaceObject wo = w.addObject(IdfObject(IddObjectType::Schedule_Constant)).get();
+    wo.setString(0, "AlwaysOn");
+    wo.setDouble(2, 1.0);
+
+    auto& newObject = energyPlusOutputRequests.emplace_back(IddObjectType::Schedule_Constant);
+    newObject.setString(0, "AlwaysOn");
+    newObject.setDouble(2, 1.0);
+
+    EXPECT_TRUE(w.getObjectByTypeAndName(IddObjectType::Schedule_Constant, "AlwaysOn"));
+    EXPECT_FALSE(openstudio::workflow::util::addEnergyPlusOutputRequest(w, newObject));
+    {
+      EXPECT_EQ(1, w.getObjectsByType(IddObjectType::Schedule_Constant).size());
+      EXPECT_TRUE(w.getObjectByTypeAndName(IddObjectType::Schedule_Constant, "AlwaysOn"));
+    }
+  }
+  {
+    // Non unique, and not present already
+    auto& newObject = energyPlusOutputRequests.emplace_back(IddObjectType::Schedule_Constant);
+    newObject.setString(0, "A New Schedule");
+    newObject.setDouble(2, 18.0);
+    EXPECT_EQ(1, w.getObjectsByType(IddObjectType::Schedule_Constant).size());
+    EXPECT_TRUE(w.getObjectByTypeAndName(IddObjectType::Schedule_Constant, "AlwaysOn"));
+    EXPECT_TRUE(openstudio::workflow::util::addEnergyPlusOutputRequest(w, newObject));
+    {
+      auto wos = w.getObjectsByType(IddObjectType::Schedule_Constant);
+      ASSERT_EQ(2, wos.size());
+      EXPECT_TRUE(w.getObjectByTypeAndName(IddObjectType::Schedule_Constant, "AlwaysOn"));
+      auto new_wo_ = w.getObjectByTypeAndName(IddObjectType::Schedule_Constant, "A New Schedule");
+      ASSERT_TRUE(new_wo_);
+      EXPECT_EQ("A New Schedule", new_wo_->getString(0).get());
+      EXPECT_TRUE(new_wo_->isEmpty(1));
+      EXPECT_EQ(18.0, new_wo_->getDouble(2).get());
+    }
+  }
+}
```

Comment:
New test for addEnergyPlusOutputRequest, specifically I'm testing what happens with unique objects. We should NOT end up with more than 1